example/lib/pages/settings/settings-page.dart

```diff
@@ -0,0 +1,299 @@
+import 'package:example/pages/settings/settings-page-state.dart';
+import 'package:floater/floater.dart';
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+
+class SettingsPage extends StatefulWidgetBase<SettingsPageState> {
+  SettingsPage() : super(() => SettingsPageState());
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: _settingsAppBar(
+        goBack: this.state.goBack,
+      ),
+      body: Padding(
+        padding: const EdgeInsets.all(20.0),
+        child: SingleChildScrollView(
+          child: Column(
+            mainAxisAlignment: MainAxisAlignment.start,
+            children: [
+              _displayInfo(),
+              _themeInfo(
```

Comment:
widgets doesn't always have to be stateless and private. for example this can be moved to another file as a stateful widget. for inspirations look at dirt

Reply:
I did stateful widget for textField

Reply:
extract it out

Reply:
I didn't get the advantage of doing like that? in my implemetation i componentize everything especially i tried to make as much as stateless widgets which give me advantages of suing stateless widget, since a page itself a stateful widget what is the advantages of extract everything into another stateful widget. could you please help me to correct my thought processes.

Reply:
because you have logic in your page that doesn't have anything to do except theme. so if you move it out you don't have to pass around callbacks and variables like that. componentization is not about making a bunch of widgets. it's about isolating funtionaliy. remember single responsibility principle